sdk/communication/Azure.Communication.CallingServer/src/Models/Events/AddParticipantsFailedEvent.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
+{
+    /// <summary>
+    /// The add participant failed event.
+    /// </summary>
+    public class AddParticipantsFailedEvent : CallingServerEventBase
+    {
+        /// <summary> Initializes a new instance of AddParticipantsFailedEvent. </summary>
+        internal AddParticipantsFailedEvent()
+        {
+            Participants = new ChangeTrackingList<CommunicationIdentifier>();
+        }
+
+        /// <summary> Initializes a new instance of AddParticipantsFailedEvent. </summary>
+        /// <param name="internalEvent">Internal Representation of the AddParticipantsFailedEvent. </param>
+        internal AddParticipantsFailedEvent(AddParticipantsFailedEventInternal internalEvent)
+        {
+            OperationContext = internalEvent.OperationContext;
+            ResultInfo = internalEvent.ResultInfo;
+            Participants = internalEvent.Participants.Select(t => CommunicationIdentifierSerializer.Deserialize(t)).ToList();
+            Type = internalEvent.Type;
+            CallConnectionId = internalEvent.CallConnectionId;
+            ServerCallId = internalEvent.ServerCallId;
+            CorrelationId = internalEvent.CorrelationId;
+        }
+
+        /// <summary> Operation context. </summary>
+        public string OperationContext { get; }
+        /// <summary> Gets the result info. </summary>
+        public ResultInformation ResultInfo { get; }
+        /// <summary> Participants failed to be added. </summary>
+        public IReadOnlyList<CommunicationIdentifier> Participants { get; }
+        /// <summary> Gets the type. </summary>
+        public AcsEventType? Type { get; }
```

Comment:
Because now we have Base abstract class has EventType, this should go away. All derived events to remove Type.

Reply:
updated all the derived event classes! thx for catching it.